src/main/java/oof/model/task/Assignment.java

```diff
@@ -24,8 +24,23 @@ public void setModuleCode(String moduleCode) {
         this.moduleCode = moduleCode;
     }
 
+    /**
+     * Returns a string from Assignment task object.
+     *
+     * @param task Assignment task object.
+     * @return String obtained from Assignment task object.
+     */
+    public String assignmentToStorageString(Assignment task) {
+        String moduleCode = getModuleCode();
+        String dateTime = task.getBy();
+        String date = dateTime.split(" ")[DATE];
+        String time = dateTime.split(" ")[TIME];
+        return "A" + DELIMITER + task.getStatusIcon() + DELIMITER + moduleCode + DELIMITER + task.getDescription()
+                + DELIMITER + date + DELIMITER + time + DELIMITER;
+    }
+
```

Comment:
Unnecessary to pass in Assignment as method is within the Assignment class thus it can directly access private attributes in its own class and protected attributes in its parent class instead of using getters. The same goes for deadline, event and todo.

Reply:
Also, it would be better if you declared an abstract method "toStorageString" in Task, thus requiring it to be implemented by all subclasses, since any class that extends Task needs to be stored in our save file.